test/TurnUITests.java

```diff
@@ -2,29 +2,99 @@
 import com.TTT.BoardPrinter;
 import com.TTT.Game;
 import com.TTT.TurnUI;
+import org.junit.Before;
 import org.junit.Test;
 
 import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
 
 import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 
 public class TurnUITests {
+
+    private Board board;
+    private Game game;
+    private BoardPrinter printer;
+
+    @Before
+    public void setUp() throws Exception {
+       board = new Board();
+       game = new Game(board);
+       printer = new BoardPrinter(System.out);
+    }
+
     @Test
     public void canAlternateTurns() {
-        TurnUI turns = new TurnUI();
-        Board board = new Board();
-        BoardPrinter printer = new BoardPrinter(System.out);
-        Game game = new Game(board);
-        ByteArrayInputStream in = new ByteArrayInputStream("1".getBytes());
+        ByteArrayInputStream in = new ByteArrayInputStream("1\n2\n".getBytes());
+        TurnUI turns = new TurnUI(printer, System.out, in);
+
+        turns.takeTurn(board, game);
+        turns.takeTurn(board, game);
+
+        assertEquals(board.markerAt(1), "O");
+    }
+
+    @Test
+    public void asksForInputAgainIfInputIsText() {
+        ByteArrayInputStream in = new ByteArrayInputStream(("invalidInput\n2").getBytes());
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        PrintStream out = new PrintStream(outputStream);
+        TurnUI turns = new TurnUI(printer, out, in);
+
+        turns.takeTurn(board, game);
+
+        assertEquals(outputStream.toString(), "Please enter a space number: Invalid input. Please enter a space number: ");
+    }
+
+    @Test
+    public void asksForInputAgainIfInputIsTooLarge() {
+        ByteArrayInputStream in = new ByteArrayInputStream(("9999999999999999999999999999\n7").getBytes());
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
```

Comment:
There is some duplication amongst the tests - could it be extracted into some small methods, which might help simplify the tests a little.